src/node.rs

```diff
@@ -315,13 +328,14 @@ where
         }
 
         if !counts.is_empty() {
+            //error is here
```

Comment:
I don't seem to understand why the usage of hashmap is needed.. what is the intention on the children here? 

If possible I would also like to remove the Hash dependency for T; and this is for sure blocking it; I'd like to find a way to avoid it, while keeping the logic the same!